### PR TITLE
Endpoint artifacts rename to deployment artifacts

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
I renamed endpoint artifacts to deployment artifacts to unify them with existing concepts. Change is breaking methods related to the retrieval of those objects.

Template update: https://github.com/zenml-io/template-e2e-batch/pull/28

This is a reborn https://github.com/zenml-io/zenml/pull/2071 which got way too out of sync with develop.

Shall go only after https://github.com/zenml-io/zenml/pull/2109

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/user-guide/advanced-guide/artifact-management/artifact-versioning.md [hunk 1]
```diff
@@ -139,7 +139,7 @@ how artifacts are linked to models:
 - `model_name`: The name of the model to link the artifact to.
 - `model_version`: The version of the model to link the artifact to.
 - `is_model_artifact`: Whether the artifact is a model artifact.
-- `is_endpoint_artifact`: Whether the artifact is an endpoint artifact.
+- `is_deployment_artifact`: Whether the artifact is an deployment artifact.
```
Comment (stefannica) [inline docs/book/user-guide/advanced-guide/artifact-management/artifact-versioning.md:142]:
```suggestion
- `is_deployment_artifact`: Whether the artifact is a deployment artifact.
```

File: examples/nlp-case/run.py [hunk 2]
```diff
@@ -67,7 +67,7 @@
 @click.option(
     "--no-cache",
     is_flag=True,
-    default=False,
+    default=True,
```
Comment (stefannica) [inline examples/nlp-case/run.py:70]: Was this intended or just a test leftover ?

File: src/zenml/artifacts/artifact_config.py [hunk 3]
```diff
@@ -57,7 +57,7 @@ def my_step() -> Annotated[
             (42), stage (ModelStages.PRODUCTION or "production"), or
             (ModelStages.LATEST or None) for the latest version (default).
         is_model_artifact: Whether the artifact is a model artifact.
-        is_endpoint_artifact: Whether the artifact is an endpoint artifact.
+        is_deployment_artifact: Whether the artifact is an deployment artifact.
```
Comment (stefannica) [inline src/zenml/artifacts/artifact_config.py:60]:
```suggestion
        is_deployment_artifact: Whether the artifact is a deployment artifact.
```

File: examples/nlp-case/run.py (hunk 2 above)
Comment (avishniakov) [reply examples/nlp-case/run.py:70]: No, it is from Safoine's template - don't know the intention, but it was just auto-refreshed

**Pull request merged**